OndselSolver/ASMTSpatialContainer.cpp

```diff
@@ -307,14 +307,18 @@ void MbD::ASMTSpatialContainer::updateFromMbD()
 	auto aAOp = mbdPart->aAOp();
 	auto vOcmO = mbdPart->qXdot()->times(mbdUnts->velocity);
 	auto omeOPO = mbdPart->omeOpO()->times(mbdUnts->omega);
+	omega3D = omeOPO;
 	auto aOcmO = mbdPart->qXddot()->times(mbdUnts->acceleration);
 	auto alpOPO = mbdPart->alpOpO()->times(mbdUnts->alpha);
 	auto& rPcmP = principalMassMarker->position3D;
 	auto& aAPp = principalMassMarker->rotationMatrix;
 	auto aAOP = aAOp->timesTransposeFullMatrix(aAPp);
+	rotationMatrix = aAOP;
 	auto rPcmO = aAOP->timesFullColumn(rPcmP);
 	auto rOPO = rOcmO->minusFullColumn(rPcmO);
+	position3D = rOPO;
```

Comment:
I'm not a fan of duplicating the pointers. This could potentially be a source of errors in the future. Perhaps it would be best to change the name of the original pointers instead. Especially since the original names are not intuitively meaningful.

Reply:
duplicating shared_ptr should be ok, right?

Reply:
Yes duplicating a std::shared_ptr is generally safe. But its building complexity and may at some point introduce coding errors.